tidy3d/components/simulation.py

```diff
@@ -908,6 +913,34 @@ def _validate_tfsf_nonuniform_grid(self) -> None:
                             f"axis, '{'xyz'[source.injection_axis]}'."
                         )
 
+    def _validate_customsourcetime(self) -> None:
+        """Make sure custom source time is not undersampled.
+        Also, make sure that all simulation.tmesh values are covered."""
+        for source in self.sources:
+            if isinstance(source.source_time, CustomSourceTime):
+                dataset = source.source_time.source_time_dataset
+                if dataset is None:
+                    continue
+                times = dataset.values.coords["t"].values
+                if (
+                    min(times) >= self.tmesh[0] + CUSTOMSOURCETIME_TOL * self.dt
```

Comment:
what is the intuition behind this choice? so the user can essentially start the time coordinates at `dt` and end them at `run_time - dt` and it should pass?